cmds/vboot/vboot.go

```diff
@@ -0,0 +1,122 @@
+package main
+
+import (
+	"crypto/sha1"
+	"crypto/sha256"
+	"flag"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"syscall"
+
+	"github.com/zaolin/go-tpm/tpm"
+	"golang.org/x/crypto/ed25519"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	tpmDevice  string = "/dev/tpm0"
+	mountPath  string = "/mnt/vboot"
+	filesystem string = "ext3"
+)
+
+func dieHard() {
+	if e1, e2, err := syscall.Syscall6(syscall.SYS_REBOOT, syscall.LINUX_REBOOT_MAGIC1, syscall.LINUX_REBOOT_MAGIC2, syscall.LINUX_REBOOT_CMD_POWER_OFF, 0, 0, 0); err != 0 {
+		log.Fatalf("a %v b %v err %v", e1, e2, err)
+	}
+}
+
+func main() {
+	var publicKey = flag.String("pubkey", "/etc/sig.pub", "A public key which should verify the signature.")
+	var pcr = flag.Uint("pcr", 12, "The pcr index used for measuring the kernel before kexec.")
+	var bootDev = flag.String("boot-device", "/dev/sda1", "The boot device which is used to kexec into a signed kernel.")
+	var linuxKernel = flag.String("kernel", "/mnt/vboot/kernel", "Kernel image file path.")
+	var linuxKernelSignature = flag.String("kernel-sig", "/mnt/vboot/kernel.sig", "Kernel image signature file path.")
+	var initrd = flag.String("initrd", "/mnt/vboot/initrd", "Initrd file path.")
+	var initrdSignature = flag.String("initrd-sig", "/mnt/vboot/initrd.sig", "Initrd signature file path.")
+	var debug = flag.Bool("debug", false, "Enables debug mode.")
+	var noTPM = flag.Bool("no-tpm", false, "Disables tpm measuring process.")
+
+	flag.Parse()
+
+	if err := os.MkdirAll(mountPath, os.ModePerm); err != nil {
+		if *debug {
+			panic(err)
+		} else {
+			dieHard()
+		}
+	}
+
+	if err := unix.Mount(*bootDev, mountPath, filesystem, unix.MS_RDONLY, ""); err != nil {
+		if *debug {
+			panic(err)
+		} else {
+			dieHard()
+		}
+	}
+
+	paths := [...]string{*publicKey, *linuxKernel, *linuxKernelSignature, *initrd, *initrdSignature}
```

Comment:
Actually I've been dinged on this kind of thing when committing to Go upstream: when I used [] they seem to want [...] since that's an array, not a slice. But who knows.